tests/smoke-nuget-directory-packages-props.yaml

```diff
@@ -0,0 +1,33 @@
+input:
+    job:
+        package-manager: nuget
+        allowed-updates:
+            - update-type: all
+        ignore-conditions:
+            - dependency-name: NuGet.Versioning
+              source: tests/smoke-nuget-directory-packages-props.yaml
+              version-requirement: '>6.2.2'
+        source:
```

Comment:
This bit confuses me a bit. Are the files going to be downloaded from GitHub?
How does one test this locally?

Reply:
I'm in the same position - trying to add a test for https://github.com/dependabot/dependabot-core/pull/6031, but haven't been able to find any documentation to explain how to accomplish this.

I've been trying to reverse engineer, but haven't been successful. This is what I've tried:
```sh
set -e 

pushd dependabot-core
bin/docker-dev-shell nuget
popd

pushd cli
go build cmd/dependabot/dependabot.go
popd

pushd smoke-tests
export LOCAL_GITHUB_ACCESS_TOKEN=<redacted>
../cli/dependabot test --updater-image dependabot/dependabot-core-development-nuget -f=tests/smoke-nuget-lockfiles.yaml -o=tests/smoke-nuget-lockfiles.yaml --cache=cache
popd
```

Reply:
We're probably missing some deeper explanations, but did you check the README in this repo? There's some information about running and generating smoke tests.

Reply:
I see. Yes, I've been there, it's not ideal and a chicken and egg situation indeed (see https://github.com/dependabot/dependabot-core/issues/7768). What I've done in the past is to change the core PR smoke tests workflow to instead download smoke tests from my smoke tests PR, instead of from the main branch.

Reply:
> I see. Yes, I've been there, it's not ideal and a chicken and egg situation indeed (see [dependabot/dependabot-core#7768](https://github.com/dependabot/dependabot-core/issues/7768)). What I've done in the past is to change the core PR smoke tests workflow to instead download smoke tests from my smoke tests PR, instead of from the main branch.

Do you happen to have a pointer or example to share?

Reply:
I did it here, for example: https://github.com/dependabot/dependabot-core/commit/e9d7025e766660d7bcc966f948068a0fb28cb0ed.